### PR TITLE
Mafia Map Fixes and Removal

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

All mafia related, i'm not talking about snowdin away mission vvvv
Snowdin now doesn't have bad atmos that gives this ugly gas turf on the catwalks.
Snowdin's airlocks now have 0 opacity so people at the bottom can see people at the top
Both Necropolis and Snowdin no longer give the fire overlay to all the people who spawn on it

## Why It's Good For The Game

Yay, the last of the map problems! Will readd other maps I cut during development due to less fixable issues when I get to them!

image

## Changelog
:cl:
fix: Fixed up Mafia maps, at least in all the ways I caught!
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
can you make a mafia subtype for necropolis lava too?

**Comment by tralezab** [comment]
I'm not entirely sure that's needed as there is normal atmos for that lava but I may

**Comment by Fikou** [comment]
the problem is that is slows down

**Comment by tralezab** [comment]
I may not remove a map on second thought, i want to try reworking the look first

File: code/modules/mob/living/carbon/carbon_update_icons.dm [hunk 1]
```diff
@@ -80,7 +80,12 @@
 
 /mob/living/carbon/update_fire(var/fire_icon = "Generic_mob_burning")
 	remove_overlay(FIRE_LAYER)
-	if(on_fire || islava(loc))
+	var/lavaburn = FALSE
+	if(islava(loc))
```
Comment (AnturK) [inline code/modules/mob/living/carbon/carbon_update_icons.dm:84]: Why is this even handled here. Awful

Comment (tralezab) [reply code/modules/mob/living/carbon/carbon_update_icons.dm:84]: t-they did it first! okay okay though, i'll see if i can tie the whole islava part to the lava turf and not update fire

File: code/game/turfs/open/lava.dm [hunk 2]
```diff
@@ -166,6 +166,8 @@
 
 				if(S && H && S.clothing_flags & LAVAPROTECT && H.clothing_flags & LAVAPROTECT)
 					return
+				else
+					C.update_fire(force = TRUE)//you will always be burning on top of lava if you're not protected
```
Comment (AnturK) [inline code/game/turfs/open/lava.dm:170]: There's already update_fire above, and there's another return below so this seems like inappropriate location ?

Comment (tralezab) [reply code/game/turfs/open/lava.dm:170]: hummm...

**Comment by tralezab** [comment]
okay, that should be everything woo!

**Pull request merged**